db/sqlinterfaces.c

```diff
@@ -3790,6 +3790,29 @@ static int bind_parameters(struct reqlogger *logger, sqlite3_stmt *stmt,
             }
             continue;
         }
+        if (p.arraylen) {
+            int flag;
+            switch (p.type) {
+            case CLIENT_INT:
+            case CLIENT_UINT:
+                //we dont have 32bit int: flag = 0; /* CARRAY_INT32; */ break;
+                flag = 1; /* CARRAY_INT64; */ break;
+            case CLIENT_REAL:
+                flag = 2; /* CARRAY_DOUBLE; */ break;
+            case CLIENT_CSTR:
+            case CLIENT_VUTF8:
+                flag = 3; /* CARRAY_TEXT; */ break;
+            default:
+                logmsg(LOGMSG_ERROR, "Unsupported type %d for carray_bind\n", p.type);
+                rc = -1;
+                return rc;
+            }
+
+            extern int sqlite3_carray_bind(sqlite3_stmt *pStmt, int idx, void *aData, int nData, int mFlags, void (*xDestroy)(void*)); 
+            rc = sqlite3_carray_bind(stmt, p.pos, p.u.p, p.arraylen, flag, SQLITE_STATIC);
+            printf("AZ: binding array for int, rc=%d, first element %ld\n", rc, *(int64_t*) p.u.p);
```

Comment:
Debug trace - going to assume you're still debugging this, so not adding more debug trace commends.